diff --git a/modules/gnode/gnode.module b/modules/gnode/gnode.module
index f37c66e..f7b5d9a 100644
--- a/modules/gnode/gnode.module
+++ b/modules/gnode/gnode.module
@@ -12,6 +12,7 @@ use Drupal\node\NodeInterface;
 use Drupal\node\Entity\NodeType;
 use Drupal\Core\Url;
 use Drupal\Core\Link;
+use Drupal\Core\Cache\Cache;
 use Drupal\Core\Access\AccessResult;
 use Drupal\Core\Database\Database;
 use Drupal\Core\Session\AccountInterface;
@@ -312,3 +313,31 @@ function gnode_node_access_records(NodeInterface $node) {
 
   return $records;
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function gnode_group_content_insert(Drupal\Core\Entity\EntityInterface $entity) {
+  // Documentation
+  $inserted_entity = $entity->getEntity();
+  $inserted_entity_type = $inserted_entity->getEntityType()->id();
+  $inserted_entity_id = $inserted_entity->id();
+
+  // Tags are based on entity type : entity id.
+  $cache_tag = $inserted_entity_type . ':' . $inserted_entity_id;
+  Cache::invalidateTags(array($cache_tag));
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_delete().
+ */
+function gnode_group_content_delete(Drupal\Core\Entity\EntityInterface $entity) {
+  // Documentation
+  $inserted_entity = $entity->getEntity();
+  $inserted_entity_type = $inserted_entity->getEntityType()->id();
+  $inserted_entity_id = $inserted_entity->id();
+
+  // Tags are based on entity type : entity id.
+  $cache_tag = $inserted_entity_type . ':' . $inserted_entity_id;
+  Cache::invalidateTags(array($cache_tag));
+}
\ No newline at end of file
